### ISSUE TITLE
UniversalSearchStore state updates interfering with search page refreshes

### ISSUE THREAD

**Issue by kroepke** [Graylog2/graylog2-server#3354]
When adding or removing a search decorator (and possibly other state changes), a component propagates new component state in a finally handler.

This interferes with other rendering, leading to 12 exceptions of this for:

```
vendor.js:48691 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the SearchPage component.
warning	@	vendor.js:48691
getInternalInstanceReadyForUpdate	@	vendor.js:12159
enqueueSetState	@	vendor.js:12306
ReactComponent.setState	@	vendor.js:26991
promise.UniversalSearchStore.search.then._this2.setState.error	@	SearchPage.jsx?./~/babel-loader/lib?cacheDirectory&extends=/home/kroepke/projects/graylog/graylog-p…:140
tryCatcher	@	bluebird.js:5215
Promise._settlePromiseFromHandler	@	bluebird.js:3238
Promise._settlePromise	@	bluebird.js:3295
Promise._settlePromise0	@	bluebird.js:3340
Promise._settlePromises	@	bluebird.js:3419
(anonymous)	@	bluebird.js:175
```

This is possibly also the cause of the "no search results" page after adding a format string decorator.

**Comment by dennisoelkers** [comment]
Refs #3355

**Comment by dennisoelkers** [comment]
Fixed in #3357.
